database/migrations/default/1750435376415_ems_person_match_auto/up.sql

```diff
@@ -0,0 +1,437 @@
+    -- todo: index these columns?
+    --
+    select id from ems__incidents e where
+    e.incident_number = ems_record.incident_number
+    and e.id !=  ems_record.id
+    and e.pcr_patient_age = ems_record.pcr_patient_age
+    and e.pcr_patient_race = ems_record.pcr_patient_race
+    and e.pcr_patient_gender = ems_record.pcr_patient_gender
+    limit 1
+    into other_matching_ems_id;
+    --
+    -- Do not proceed if multiple identical PCRs 
+    --
+    IF other_matching_ems_id IS NOT NULL THEN
+        raise debug 'Multiple EMS patients with same demographic characteristcs found - aborting person match.';
+        return NULL;
+    END IF;
+    --
+    -- Query for matching person records
+    --
+    SELECT
+        COALESCE(array_agg(p.id), ARRAY[]::integer[])
+    FROM
+        people_list_view p
+        left join lookups.drvr_ethncty ethncty_lkp on ethncty_lkp.id = p.prsn_ethnicity_id
+        left join lookups.gndr gndr_lkp on gndr_lkp.id = p.prsn_gndr_id
+    WHERE
+        p.crash_pk = ems_record.crash_pk
+        and p.prsn_age = ems_record.pcr_patient_age
+        AND gndr_lkp.label ilike ems_record.pcr_patient_gender
+        AND ems_record.pcr_patient_race ilike CONCAT('%', ethncty_lkp.label, '%')
+        into matching_person_ids;
+
+    raise debug 'Found % matching people records', array_length(matching_person_ids, 1);
+    return NULLIF(matching_person_ids, '{}');
+END;
+$$ LANGUAGE plpgsql;
+
+--
+-- New trigger that handles matching using an event-driven approach
+--
+CREATE
+OR REPLACE FUNCTION public.ems_update_handle_record_match_event() RETURNS trigger LANGUAGE plpgsql AS $function$
+DECLARE
+    matching_person_ids INTEGER[];
+    matching_person_record RECORD;
+BEGIN
+    raise debug '**function: ems_update_handle_record_match_event for EMS ID: % **, event: %', NEW.id, NEW._match_event_name;
+    --
+    -- Only continue if a valid _match_event_name is provided
+    --
+     IF NEW._match_event_name is null then
+        return new;
+    ELSIF NEW._match_event_name not in (
+        'match_crash_by_automation',
+        'match_person_by_manual_qa',
+        'reset_crash_match',
+        'sync_crash_pk_on_person_match',
+        'unmatch_crash_by_manual_qa',
+        'unmatch_person_by_manual_qa',
+        'update_matched_crash_ids'
+    ) then
+        RAISE EXCEPTION 'Invalid _match_event_name: `%`', NEW._match_event_name
+        USING HINT = 'Check this function definition for allowed _match_event_name values',
+            -- code 23514 is check_violation
+            ERRCODE = '23514';
+    END IF;
+    --
+    -- `unmatch_crash_by_manual_qa` event is set from the VZE using the
+    -- "Match not found" dropdown menu option
+    --
+    IF NEW._match_event_name = 'unmatch_crash_by_manual_qa' then
+        NEW.crash_match_status = 'unmatched_by_manual_qa'; 
+        NEW.crash_pk = NULL;
+        NEW.person_match_status = 'unmatched_by_manual_qa';
+        NEW.matched_person_ids = NULL;
```

Comment:
so this column actually shouldnt stay up to date behind the scenes like i said in my comment above, bc if crash id is manually unmatched then this will nullify.